js/spec/nested-file-test.js

```diff
@@ -0,0 +1,73 @@
+const path = require('path');
+const assert = require('assert');
+const fse = require('fs-extra');
+
+const { DEBOUNCE, TIMEOUT_PER_STEP, WORKDIR: workDir, sleep } = require('./common');
+
+const nsfw = require('../src');
+
+/**
+ * Run this function to register the test
+ * @returns {void}
+ */
+module.exports = function() {
+  it('can listen for the creation of a deeply nested file', async function () {
+    const folders = 'a_very_deep_tree'.split(''); // ['a', '_', ...]
+    const file = 'a_file.txt';
+
+    // Resolve a promise as soon as all events have been found
+    let done; const promise = new Promise(resolve => {
+      done = resolve;
+    });
+
+    /**
+     * We will remove or pop entries as we find them.
+     * This set should be empty at the end of the test.
+     * @type {Set<string>}
+     */
+    const events = new Set();
+    let directory = workDir; for (const folder of folders) {
+      directory = path.join(directory, folder);
+      events.add(directory);
+    }
+    const filePath = path.join(directory, file);
+    events.add(filePath);
+
+    function findEvent(element) {
+      if (element.action === nsfw.actions.CREATED) {
+        const file = path.join(element.directory, element.file);
+        if (events.has(file)) {
+          events.delete(file);
+          if (events.size === 0) {
+            done();
+          }
+        }
+      }
+    }
+
+    let watch = await nsfw(
+      workDir,
+      events => events.forEach(findEvent),
+      { debounceMS: DEBOUNCE }
+    );
+
+    try {
+      await watch.start();
+      await sleep(TIMEOUT_PER_STEP);
+      await fse.mkdirp(directory);
+      await fse.close(await fse.open(filePath, 'w'));
+      await Promise.race([
+        sleep(folders.length * 500 * 1.5),
```

Comment:
Is this the timer that's holding on for dear life? Maybe we should cancel the timeout if it loses the race.

Reply:
I checked with `why-is-node-running` after the test and the event loop was empty. Also that sleep is not going on for an hour, which is the time it took for CI to be cancelled... Also, is is possible to cancel promisified timeouts? I didn't bother for the aforementioned reasons.

Reply:
And what confuses even more is that on two different Linux machines the test runs fine, only on GH Actions does it hang...